### PR TITLE
Fix tracer dnf plugin location

### PR DESCRIPTION
**Pull request by pcreech**

### CODE REVIEW

**Comment by ianballou** [comment]
I confirmed that moving tracer_upload to dnf-plugins fixed the Tracer issues.

**Comment by evgeni** [comment]
[test rpm]

File: packages/client/katello-host-tools/katello-host-tools.spec [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 %if %{dnf_install}
 %global python_libdir %{python3_sitelib}
-%global plugins_dir %{python3_sitelib}/dnf_plugins
+%global plugins_dir %{python3_sitelib}/dnf-plugins
```
Comment (evgeni) [inline packages/client/katello-host-tools/katello-host-tools.spec:15]: this is in line with the old spec of 3.5.7:

https://github.com/theforeman/foreman-packaging/blob/f1ce67c948d47772d769417e753fa2dd0fe411f7/packages/client/katello-host-tools/katello-host-tools.spec#L152

**Comment by evgeni** [comment]
[test rpm]

**Comment by evgeni** [comment]
So I went looking whether there were any other unexpected changes between 3.5.7 and 4.2.3-2 (on el8 only, I am lazy!):

the `katello-agent` pkg looks sane, the changes are mostly checksums and timestamps (as those obviously change) and the explicit addition of a few folders (previously the pkg only contained the files in the folders, which works, but oh well) plus the creation of `/var/cache/katello-agent` which previously was wrongly in `katello-host-tools`:
```console
% rpmdiff ~/Downloads/katello-agent-* 
removed     REQUIRES config(katello-agent) = 3.5.7-5.el8
removed     REQUIRES katello-host-tools = 3.5.7-5.el8
added       REQUIRES config(katello-agent) = 4.2.3-2.el8
added       REQUIRES katello-host-tools = 4.2.3-2.el8
removed     PROVIDES config(katello-agent) = 3.5.7-5.el8
removed     PROVIDES katello-agent = 3.5.7-5.el8
added       PROVIDES config(katello-agent) = 4.2.3-2.el8
added       PROVIDES katello-agent = 4.2.3-2.el8
..........T /etc/gofer/plugins/katello.conf
added       /usr/lib/python3.6/site-packages/katello/agent
..........T /usr/lib/python3.6/site-packages/katello/agent/__init__.py
added       /usr/lib/python3.6/site-packages/katello/agent/__pycache__
..5.......T /usr/lib/python3.6/site-packages/katello/agent/__pycache__/__init__.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/__pycache__/__init__.cpython-36.pyc
added       /usr/lib/python3.6/site-packages/katello/agent/goferd
..........T /usr/lib/python3.6/site-packages/katello/agent/goferd/__init__.py
added       /usr/lib/python3.6/site-packages/katello/agent/goferd/__pycache__
..5.......T /usr/lib/python3.6/site-packages/katello/agent/goferd/__pycache__/__init__.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/goferd/__pycache__/__init__.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/goferd/__pycache__/plugin.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/goferd/__pycache__/plugin.cpython-36.pyc
..........T /usr/lib/python3.6/site-packages/katello/agent/goferd/plugin.py
added       /usr/lib/python3.6/site-packages/katello/agent/pulp
..........T /usr/lib/python3.6/site-packages/katello/agent/pulp/__init__.py
added       /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/__init__.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/__init__.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/dispatcher.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/dispatcher.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/handler.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/handler.cpython-36.pyc
S.5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/libdnf.cpython-36.opt-1.pyc
S.5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/libdnf.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/libyum.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/libyum.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/report.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/__pycache__/report.cpython-36.pyc
..........T /usr/lib/python3.6/site-packages/katello/agent/pulp/dispatcher.py
..........T /usr/lib/python3.6/site-packages/katello/agent/pulp/handler.py
S.5.......T /usr/lib/python3.6/site-packages/katello/agent/pulp/libdnf.py
..........T /usr/lib/python3.6/site-packages/katello/agent/pulp/libyum.py
..........T /usr/lib/python3.6/site-packages/katello/agent/pulp/report.py
added       /var/cache/katello-agent
```

`katello-host-tools` also looks fine, `scripts.py` is a new thing in 4.x (via https://github.com/Katello/katello-host-tools/commit/8c5b6c1ece2ef7ff76d781a2d5a41443c9ea0845), but one thing wonders me: 3.5.7 did not actually have any binaries in `/usr/(s)?bin`?! Looking at the old spec (https://github.com/theforeman/foreman-packaging/blob/f1ce67c948d47772d769417e753fa2dd0fe411f7/packages/client/katello-host-tools/katello-host-tools.spec) it installed those only for non-dnf installs (= EL6/7 and SUSE), which seems odd to me but is it because the upload there is completely handled inside dnf/sub-man and does not require any binaries?!
```console
% rpmdiff ~/Downloads/katello-host-tools-3.5.7-5.el8.noarch.rpm ~/Downloads/katello-host-tools-4.2.3-2.el8.noarch.rpm
added       REQUIRES /usr/libexec/platform-python  
removed     PROVIDES katello-host-tools = 3.5.7-5.el8
added       PROVIDES katello-host-tools = 4.2.3-2.el8
added       PROVIDES python3.6dist(katello-host-tools) = 4.2.3
added       PROVIDES python3dist(katello-host-tools) = 4.2.3
removed     OBSOLETES katello-host-tools-fact-plugin < 3.5.7-5.el8
added       OBSOLETES katello-host-tools-fact-plugin < 4.2.3-2.el8
..........T /usr/lib/python3.6/site-packages/katello
..........T /usr/lib/python3.6/site-packages/katello/__init__.py
added       /usr/lib/python3.6/site-packages/katello/__pycache__
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/__init__.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/__init__.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/constants.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/constants.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/enabled_report.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/enabled_report.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/packages.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/packages.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/repos.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/repos.cpython-36.pyc
added       /usr/lib/python3.6/site-packages/katello/__pycache__/scripts.cpython-36.opt-1.pyc
added       /usr/lib/python3.6/site-packages/katello/__pycache__/scripts.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/uep.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/uep.cpython-36.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/utils.cpython-36.opt-1.pyc
..5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/utils.cpython-36.pyc
..........T /usr/lib/python3.6/site-packages/katello/constants.py
..........T /usr/lib/python3.6/site-packages/katello/enabled_report.py
..........T /usr/lib/python3.6/site-packages/katello/packages.py
..........T /usr/lib/python3.6/site-packages/katello/repos.py
added       /usr/lib/python3.6/site-packages/katello/scripts.py
..........T /usr/lib/python3.6/site-packages/katello/uep.py
..........T /usr/lib/python3.6/site-packages/katello/utils.py
added       /usr/lib/python3.6/site-packages/katello_host_tools-4.2.3-py3.6.egg-info
added       /usr/lib/python3.6/site-packages/katello_host_tools-4.2.3-py3.6.egg-info/PKG-INFO
added       /usr/lib/python3.6/site-packages/katello_host_tools-4.2.3-py3.6.egg-info/SOURCES.txt
added       /usr/lib/python3.6/site-packages/katello_host_tools-4.2.3-py3.6.egg-info/dependency_links.txt
added       /usr/lib/python3.6/site-packages/katello_host_tools-4.2.3-py3.6.egg-info/entry_points.txt
added       /usr/lib/python3.6/site-packages/katello_host_tools-4.2.3-py3.6.egg-info/top_level.txt
added       /usr/sbin/katello-enabled-repos-upload
added       /usr/sbin/katello-package-upload
removed     /usr/share/doc/katello-host-tools
removed     /usr/share/doc/katello-host-tools/LICENSE
added       /usr/share/licenses/katello-host-tools
added       /usr/share/licenses/katello-host-tools/LICENSE
removed     /var/cache/katello-agent
```

and then `k-h-t-tracer` that clearly shows the movement of the dnf plugins folder
```console
% rpmdiff ~/Downloads/katello-host-tools-tracer-3.5.7-5.el8.noarch.rpm ~/Downloads/katello-host-tools-tracer-4.2.3-2.el8.noarch.rpm
removed     REQUIRES config(katello-host-tools-tracer) = 3.5.7-5.el8
removed     REQUIRES katello-host-tools = 3.5.7-5.el8
added       REQUIRES config(katello-host-tools-tracer) = 4.2.3-2.el8
added       REQUIRES katello-host-tools = 4.2.3-2.el8
removed     PROVIDES config(katello-host-tools-tracer) = 3.5.7-5.el8
removed     PROVIDES katello-host-tools-tracer = 3.5.7-5.el8
added       PROVIDES config(katello-host-tools-tracer) = 4.2.3-2.el8
added       PROVIDES katello-host-tools-tracer = 4.2.3-2.el8
S.5.......T /etc/cron.d/katello-tracer-upload
..........T /etc/dnf/plugins/tracer_upload.conf
removed     /usr/lib/python3.6/site-packages/dnf-plugins/__pycache__/tracer_upload.cpython-36.opt-1.pyc
removed     /usr/lib/python3.6/site-packages/dnf-plugins/__pycache__/tracer_upload.cpython-36.pyc
removed     /usr/lib/python3.6/site-packages/dnf-plugins/tracer_upload.py
added       /usr/lib/python3.6/site-packages/dnf_plugins/__pycache__/tracer_upload.cpython-36.opt-1.pyc
added       /usr/lib/python3.6/site-packages/dnf_plugins/__pycache__/tracer_upload.cpython-36.pyc
added       /usr/lib/python3.6/site-packages/dnf_plugins/tracer_upload.py
S.5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/tracer.cpython-36.opt-1.pyc
S.5.......T /usr/lib/python3.6/site-packages/katello/__pycache__/tracer.cpython-36.pyc
S.5.......T /usr/lib/python3.6/site-packages/katello/tracer.py
..........T /usr/sbin/katello-tracer-upload
```

and now, looking at the new package in this PR, we see the plugins being moved back correctly:
```console
% rpmdiff ~/Downloads/katello-host-tools-tracer-4.2.3-2.el8.noarch.rpm ~/Downloads/katello-host-tools-tracer-4.2.3-3.el8.noarch.rpm
removed     REQUIRES config(katello-host-tools-tracer) = 4.2.3-2.el8
removed     REQUIRES katello-host-tools = 4.2.3-2.el8
added       REQUIRES config(katello-host-tools-tracer) = 4.2.3-3.el8
added       REQUIRES katello-host-tools = 4.2.3-3.el8
removed     PROVIDES config(katello-host-tools-tracer) = 4.2.3-2.el8
removed     PROVIDES katello-host-tools-tracer = 4.2.3-2.el8
added       PROVIDES config(katello-host-tools-tracer) = 4.2.3-3.el8
added       PROVIDES katello-host-tools-tracer = 4.2.3-3.el8
..........T /etc/cron.d/katello-tracer-upload
..........T /etc/dnf/plugins/tracer_upload.conf
added       /usr/lib/python3.6/site-packages/dnf-plugins/__pycache__/tracer_upload.cpython-36.opt-1.pyc
added       /usr/lib/python3.6/site-packages/dnf-plugins/__pycache__/tracer_upload.cpython-36.pyc
added       /usr/lib/python3.6/site-packages/dnf-plugins/tracer_upload.py
removed     /usr/lib/python3.6/site-packages/dnf_plugins/__pycache__/tracer_upload.cpython-36.opt-1.pyc
removed     /usr/lib/python3.6/site-packages/dnf_plugins/__pycache__/tracer_upload.cpython-36.pyc
removed     /usr/lib/python3.6/site-packages/dnf_plugins/tracer_upload.py
..........T /usr/lib/python3.6/site-packages/katello/__pycache__/tracer.cpython-36.opt-1.pyc
..........T /usr/lib/python3.6/site-packages/katello/__pycache__/tracer.cpython-36.pyc
..........T /usr/sbin/katello-tracer-upload
```

So, TL;DR: this PR fixes the tracer issue, but I wonder if the scripts changes in tools are correct.

**Review by evgeni — APPROVED** [review]

**Comment by ehelms** [comment]
Subscription manager handles those two functions for us , there are configuration options in rhsm.conf

**Comment by evgeni** [comment]
> Subscription manager handles those two functions for us , there are configuration options in rhsm.conf

https://github.com/theforeman/foreman-packaging/pull/9305
